storage/inmemory/src/lib.rs

```diff
@@ -32,8 +32,8 @@ impl<'i> Iterator for PrefixIter<'i> {
 pub struct StorageMaps(Vec<Map>);
 
 impl backend::ReadOps for StorageMaps {
-    fn get(&self, idx: DbIndex, key: &[u8]) -> storage_core::Result<Option<&[u8]>> {
-        Ok(self.0[idx.get()].get(key).map(AsRef::as_ref))
+    fn get(&self, idx: DbIndex, key: &[u8]) -> storage_core::Result<Option<Cow<[u8]>>> {
+        Ok(self.0[idx.get()].get(key).map(|p| p[..].into()))
```

Comment:
```suggestion
        Ok(self.0.get(idx.get()).and_then(|m| m.get(key).map(|p| p.into())))
```
Am I missing the point of `[..]` here? And wouldn't it be better to to use the `get` method instead of potentially panicking `[]`?

Reply:
The key-value maps should be initialised at the beginning, and the storage layer statically enforces the existence. Panicking is the right thing to do here, otherwise we mask potential issues with the backend not maintaining invariants properly.

Reply:
@iljakuklic Got it!

Reply:
`[..]` doesn't panic unless you dereference the slice coming out of it incorrectly. The primitive `[..]` is an idiomatic way to convert something to a slice. Even if the size is zero, it won't panic.

The reason I chose `[..]` over as_ref() is because there's type inference with as_ref(), which makes into() fail. So we have to convert to a slice that infers the type.